qiita_pet/handlers/preprocessing_handlers.py

```diff
@@ -13,19 +13,12 @@ class PreprocessHandler(BaseHandler):
     def post(self):
         study_id = int(self.get_argument('study_id'))
         prep_template_id = int(self.get_argument('prep_template_id'))
-        raw_data = PrepTemplate(prep_template_id).artifacts
+        raw_data = PrepTemplate(prep_template_id).artifact
         param_id = int(self.get_argument('preprocessing_parameters_id'))
 
-        # Get the preprocessing parameters
-        if raw_data.filetype in ('FASTQ', 'per_sample_FASTQ'):
-            param_constructor = PreprocessedIlluminaParams
-        elif raw_data.filetype in ('FASTA', 'SFF'):
-            param_constructor = Preprocessed454Params
-        else:
-            raise ValueError('Unknown filetype')
+        user_id = self.current_user.id
 
-        job_id = submit(self.current_user.id, preprocessor, study_id,
-                        prep_template_id, param_id, param_constructor)
+        job_id = submit(user_id, preprocessor, user_id, raw_data.id, param_id)
```

Comment:
This `submit` function is a bit confusing.  Why does `user_id` have to be passed in twice?

Also the function signature for [`submit`](https://github.com/josenavas/QiiTa/blob/plugin-connect-tgp/qiita_ware/context.py) is really confusing.  If you think it is a good idea, I can open up an issue/PR to add numpy docs to fix this.